larray/core.py

```diff
@@ -8761,6 +8774,10 @@ def split_axis(self, axis, sep='_', names=None):
         names : list of names, optional
             names of resulting axes. Defaults to split the combined axis name
             using the given delimiter string.
+        regex : str, optional
+            use regex instead of simple delimiter to split the axis.
+            Names of new axes must be provided.
```

Comment:
Regex for labels and delimiter for names if `names` is string or axis name if `names` is None.